From bc4535159f47d884ec2f7d1bddefd3e7fd767cdc Mon Sep 17 00:00:00 2001
From: Mathew McBride <matt@traverse.com.au>
Date: Wed, 3 Jul 2024 09:33:01 +1000
Subject: [PATCH 5/9] scripts: lan_handler: create flag files for lan-start and
 ipv4_4-status events

Due to a long standing race condition issue, the ipv4_4-status handler
is not called. A modified brlan0_check.sh in meta-cmf-traverse will
try to detect and workaround this issue

Change-Id: I973fcdd05d483034b2b9b349a892dc68de35216c
---
 source/scripts/init/service.d/lan_handler.sh | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/source/scripts/init/service.d/lan_handler.sh b/source/scripts/init/service.d/lan_handler.sh
index 1c4f862..104f848 100755
--- a/source/scripts/init/service.d/lan_handler.sh
+++ b/source/scripts/init/service.d/lan_handler.sh
@@ -132,6 +132,17 @@ if [ "$1" = "lan-stop" ] && [ "$2" = "NULL" ] ; then
 fi
 #echo "lan_handler called with $1 $2" > /dev/console
 
+# Used by brlan0_check.sh to workaround LAN issue
+if [ "$1" = "lan-start" ]; then
+    touch /tmp/utopia-lan-started
+fi
+if [ "$1" = "lan-stop"]; then
+    rm /tmp/utopia-lan-started || :
+fi
+if [ "$1" = "ipv4_4-status"] && [ "$2" = "up" ]; then
+    touch /tmp/utopia-ipv4-4-up
+fi
+
 case "$1" in
    ${SERVICE_NAME}-start)
       service_start
-- 
2.51.2

